pyazul/validate.py

```diff
@@ -15,32 +63,47 @@ def datavault_create(data):
 def datavault_delete(data):
     required = {
         'DataVaultToken': data.get('DataVaultToken', ''),
-        'TrxType': 'DELETE',
+        'TrxType': r'DELETE',
     }
 
     return required
 
 
 def sale_transaction(data):
-    required = {
-        'CardNumber': data.get('CardNumber', ''),
-        'Expiration': data.get('Expiration', ''),
-        'CVC': data.get('CVC', ''),
-        'PosInputMode': data.get('PosInputMode', 'E-Commerce'),
-        'TrxType': 'Sale',
-        'Payments': data.get('Payments', '1'),
-        'Plan': data.get('Plan', '0'),
-        'Amount': str(utils.clean_amount(data.get('Amount', 0))),
-        'Itbis': utils.clean_amount(data.get('Itbis', 0)),
-        'CurrencyPosCode': data.get('CurrencyPosCode', ''),
-        'AcquirerRefData': data.get('AcquirerRefData', '1'),
-        'CustomerServicePhone': data.get('CustomerServicePhone', ''),
-        'OrderNumber': data.get('OrderNumber', ''),
-        'EcommerceURL': data.get('EcommerceURL', ''),
-        'CustomOrderID': data.get('CustomOrderID', ''),
+    validation_rule_sets = {
+        'CardNumber': (check_exists_in(data), check_not_empty,),
+        'Expiration': (check_exists_in(data),),
+        'CVC': (check_exists_in(data),),
+        'PosInputMode': (
+            check_exists_in(data),
+            check_exists_in(['E-Commerce']),
+        ),
+        'TrxType': (check_exists_in(data), check_exists_in(['Sale']),),
```

Comment:
```suggestion
        'TrxType': 'Sale',
```
'TrxType' value must be hardcoded since this will never change for a sale transaction.

Reply:
in this case it should not be validated here. This method only validates, it doesn't remap nor returns

Reply:
Oh, that's true. Let's just remove it and update the dictionary on main.py with the transaction types after being validated.